src/core/useCases/feed/feed.spec.ts

```diff
@@ -45,95 +47,137 @@ describe('Feed', () => {
-    const nextFilters: FeedState['filters'] = {
+    const nextPosition: PositionState['position'] = {
       cityName: 'Nantes',
       center: { lat: 2, lng: 3 },
       zoom: 12,
     }
-    store.dispatch(publicActions.setFilters(nextFilters))
-
-    expect(selectFeedIsFetching(store.getState())).toEqual(true)
+    store.dispatch(positionActions.setPosition(nextPosition))
 
     feedGateway.retrieveFeedItems.resolveDeferredValue()
     await store.waitForActionEnd()
 
-    expect(selectFeedIsFetching(store.getState())).toEqual(false)
+    expect(feedGateway.retrieveFeedItems).toHaveBeenCalledTimes(0)
   })
 
   it(`
-    Given feed request is idle
-    When no action is trigger by user
-    Then feed request should still be idle
-  `, () => {
+    Given the feed saga has initialized
+    When user set position filters
+    Then items should be fetched
+  `, async () => {
     const feedGateway = new TestFeedGateway()
+    feedGateway.retrieveFeedItems.mockDeferredValueOnce({ items: [], nextPageToken: null })
     const store = configureStoreWithFeed({ dependencies: { feedGateway } })
+    const nextPosition: PositionState['position'] = {
+      cityName: 'Nantes',
+      center: { lat: 2, lng: 3 },
+      zoom: 12,
+    }
 
-    expect(selectFeedIsIdle(store.getState())).toEqual(true)
+    store.dispatch(publicActions.init())
+
+    store.dispatch(positionActions.setPosition(nextPosition))
+
+    feedGateway.retrieveFeedItems.resolveDeferredValue()
+    await store.waitForActionEnd()
+
+    expect(feedGateway.retrieveFeedItems).toHaveBeenCalledTimes(1)
   })
 
   it(`
-    Given feed request is idle
-    When user retrieve feed successfully
-    Then feed request should not be idle
+    Given the feed saga has been cancelled
+    When user set position filters
+    Then items should not be fetched
   `, async () => {
     const feedGateway = new TestFeedGateway()
     feedGateway.retrieveFeedItems.mockDeferredValueOnce({ items: [], nextPageToken: null })
     const store = configureStoreWithFeed({ dependencies: { feedGateway } })
+    const nextPosition: PositionState['position'] = {
+      cityName: 'Nantes',
+      center: { lat: 2, lng: 3 },
+      zoom: 12,
+    }
+    store.dispatch(publicActions.init())
 
-    store.dispatch(publicActions.retrieveFeed())
+    store.dispatch(positionActions.setPosition(nextPosition))
 
     feedGateway.retrieveFeedItems.resolveDeferredValue()
     await store.waitForActionEnd()
 
-    expect(selectFeedIsIdle(store.getState())).toEqual(false)
+    expect(feedGateway.retrieveFeedItems).toHaveBeenCalledTimes(1)
+
+    feedGateway.retrieveFeedItems.mockDeferredValueOnce({ items: [], nextPageToken: null })
+
+    store.dispatch(publicActions.cancel())
+
+    store.dispatch(positionActions.setPosition(nextPosition))
+
+    feedGateway.retrieveFeedItems.resolveDeferredValue()
+    await store.waitForActionEnd()
+
+    expect(feedGateway.retrieveFeedItems).toHaveBeenCalledTimes(1)
```

Comment:
Il faudrait en plus mettre une autre valeur à setPosition pour s'assurer qu'on trigger bien un changement de position

```suggestion
    store.dispatch(publicActions.cancel())

    store.dispatch(positionActions.setPosition(nextPosition))

    feedGateway.retrieveFeedItems.resolveDeferredValue()
    await store.waitForActionEnd()

    expect(feedGateway.retrieveFeedItems).toHaveBeenCalledTimes(1)
```